### DISCUSSION TITLE
Add a way to use a revocable API key instead of email and password

### DISCUSSION THREAD

**Discussion by Funkatronics** [Genymobile/feedback#76] [Genymotion SaaS] [answered]
Hello,

It would be great if there was a way to use Genymotion SaaS from the command line by using an API key that can be generated (and revoked) through the cloud.geny.io dashboard. This would be more secure than using the email and password for our account since an API key could not be used to login to our account, and could be easily revoked and reissued if it was ever compromised.

**Comment by agavignet** [comment, marked as answer]
Hi @Funkatronics,

Thanks for your feedback, I got a good news for you: we're working on it, should be available by the end of the year.
Are you using gmsaas CLI or the HTTP API directly?

Adrien

**Reply by Funkatronics** [reply]
Oh that's great to hear! we are using the gmsaas CLI (via github actions to run our android ui automation tests)

**Comment by Funkatronics** [comment]
Hi @agavignet checking in on this. Will this feature be available soon?

**Reply by agavignet** [reply]
Hi @Funkatronics thanks for your patience, this is live now, you can read our blog post about it https://www.genymotion.com/blog/api-tokens-for-genymotion-saas/
